### PR TITLE
Convert deadTServers and badTServers tables to DataTables

### PR DESCRIPTION
**Pull request by DomGarguilo**
This PR:
* Converts `deadtservers` and `badtservers` to DataTables
* Add a title and caption to the online tservers table
* Conform to eslint formatting

Here is a screenshot of the current `/tservers` page:
image

### CODE REVIEW

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js [hunk 1]
```diff
@@ -183,29 +112,155 @@ function refreshDeadTServersTable() {
...
+            // reset background of each row
+            $(row).css('background-color', '');
+
+            // return if the current row's tserver is not recovering
+            if (!recoveryList.includes(data.hostname)) {
+                return;
+            }
+
+            // only show the caption if we know there are rows in the tservers table
+            $('#recovery-caption').show();
```
Comment (milleruntime) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js:198]: You could put this in an else above.

Comment (milleruntime) [reply server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js:198]: Oh wait nvm JSlint probably told you not to.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tservers.ftl [hunk 2]
```diff
@@ -25,9 +25,9 @@
       </div>
       <div class="row">
         <div class="col-xs-12">
-          <table id="badtservers" class="table table-bordered table-striped table-condensed" style="display: none;">
```
Comment (milleruntime) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/templates/tservers.ftl:28]: So how are you hiding the table now? I didn't see a hide anywhere. I think with the other datatables I had to hide the div around the table.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js [hunk 3]
```diff
@@ -16,165 +16,94 @@
...
+    ajaxReloadTable(badTServersTable);
 
-    var date = new Date(val.lastStatus);
-    date = date.toLocaleString().split(' ').join('&nbsp;');
-    items.push(createRightCell(val.lastStatus, date));
-    items.push(createRightCell(val.status, val.status));
-    items.push(createRightCell('', '<a href="javascript:clearDeadTServers(\'' +
-      val.server + '\');">clear</a>'));
+    // Only show the table if there are non-empty rows
+    if ($('#badtservers tbody .dataTables_empty').length) {
```
Comment (milleruntime) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js:84]: Nice! Sorry I missed this, ignore my other comment.

**Review by milleruntime — APPROVED** [review]
Looks good. Just wondering if you will get flashing of the new tables with moving the hiding from the html to the JS.

**Comment by DomGarguilo** [comment]
> Looks good. Just wondering if you will get flashing of the new tables with moving the hiding from the html to the JS.

No, it looks like the hidden tables don't flash on screen with autorefresh on. I made sure the highlighted rows for recovering tservers still works (which it does) but noticed that when the caption is displayed, it will flash out then back in on each autorefresh. Not sure if that is something worth looking into further or not.

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js (hunk 1 above)
Comment (DomGarguilo) [reply server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js]: Looks like jslint doesn't complain about the changes made in [086031e](https://github.com/apache/accumulo/pull/2685/commits/086031ecff58f4c089efb9eaa257218723a1553a)

**Comment by DomGarguilo** [comment]
> when the caption is displayed, it will flash out then back in on each autorefresh.

Fixed this in [78e613b](https://github.com/apache/accumulo/pull/2685/commits/78e613bf4dd4048e001c85230e25ad7fc375a61e).

File: server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js [hunk 4]
```diff
@@ -16,165 +16,115 @@
...
-
-        // highlight current row
-        console.log('Highlighting row index:' + index + ' tserver:' + data.hostname);
-        $(row).css('background-color', 'gold');
-      }
-    });
-    refreshTServers();
-});
+function serverIsInRecoveryList(server) {
+    return recoveryList.includes(server.hostname);
```
Comment (milleruntime) [inline server/monitor/src/main/resources/org/apache/accumulo/monitor/resources/js/tservers.js:36]: Cool!

**Review by milleruntime — APPROVED** [review]

**Pull request merged**